### PR TITLE
adding redis persistance

### PR DESCRIPTION
**Pull request by alanvoss**
[Card](https://rentpath.leankit.com/card/559711262)

a lot of this PR looks to be big changes, but really was a product of needing to serialize and deserialize a set of `Map`s, and doing so required either un-`atom`ifying everything (`Poison.decode!` does that by default, per JSON conventions) or doing some nasty data manipulation after the fact.

### CODE REVIEW

**Comment by michaelstalker** [comment]
👍

**Comment by michaelstalker** [comment]
Would it be a good idea to move all the Redis interaction into a separate namespace? That would help isolate the app's interaction with the external database. I know we need to get this project wrapped up, but I thought I'd raise the question anyway.

File: config/prod.exs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+use Mix.Config
+
+config :queue_bot, :redis_client,
+  use_redis: true
+
+config :exredis,
+  host: "redis",
+  port: 6379,
+  db: 8,
```
[the remark this answers is not in the document]
Comment (alanvoss) [reply config/prod.exs]: No need to do this, Pasha, because this is in AWS in a special deploy, outside of our infrastructure.

**Comment by alanvoss** [comment]
@michaelstalker I'll look into that after the next big thing.

I want to get the major pieces into places.

**Pull request merged**